subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_spi_stm32.c

```diff
@@ -0,0 +1,730 @@
+			.dest_burst_length = 1,   /* SINGLE transfer */                            \
+			.channel_priority = STM32_DMA_CONFIG_PRIORITY(                             \
+				DT_DMAS_CELL_BY_NAME(id, dir, channel_config)),                    \
+			.dma_callback = dma_callback,                                              \
+			.block_count = 2,                                                          \
+	},                                                                                         \
+	.fifo_threshold =                                                                          \
+		STM32_DMA_FEATURES_FIFO_THRESHOLD(DT_DMAS_CELL_BY_NAME(id, dir, features)),
+
+#define STM32_SPI_INIT(id)                                                                         \
+	PINCTRL_DT_DEFINE(id);                                                                     \
+	static const struct stm32_pclken pclken[] = STM32_DT_CLOCKS(id);                           \
+                                                                                                   \
+	static struct ec_host_cmd_spi_cfg ec_host_cmd_spi_cfg = {                                  \
+		.spi = (SPI_TypeDef *)DT_REG_ADDR(id),                                             \
+		.pclken = pclken,                                                                  \
+		.pclk_len = DT_NUM_CLOCKS(id),                                                     \
+		.pcfg = PINCTRL_DT_DEV_CONFIG_GET(id),                                             \
+	};                                                                                         \
+                                                                                                   \
+	static struct dma_stream dma_rx = {SPI_DMA_CHANNEL_INIT(id, rx, RX, PERIPHERAL, MEMORY)};  \
+	static struct dma_stream dma_tx = {SPI_DMA_CHANNEL_INIT(id, tx, TX, MEMORY, PERIPHERAL)}
+
+STM32_SPI_INIT(DT_CHOSEN(zephyr_host_cmd_spi_backend));
+
+#define EC_HOST_CMD_SPI_DEFINE(_name)                                                              \
+	static struct ec_host_cmd_spi_ctx _name##_hc_spi = {                                       \
+		.dma_rx = &dma_rx,                                                                 \
+		.dma_tx = &dma_tx,                                                                 \
+		.spi_config = &ec_host_cmd_spi_cfg,                                                \
+	};                                                                                         \
+	struct ec_host_cmd_backend _name = {                                                       \
+		.api = &ec_host_cmd_api,                                                           \
+		.ctx = (struct ec_host_cmd_spi_ctx *)&_name##_hc_spi,                              \
+	}
+
+static inline uint32_t dma_source_addr(SPI_TypeDef *spi)
+{
+#if DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_host_cmd_spi_backend), st_stm32h7_spi)
```

Comment:
Is this for a difference in the API from H7 to other STM32? Maybe you could check the architecture instead, how is it done in other drivers?

Reply:
This way of doing is the one I suggested and based on existing behavior on STM32 SPI driver.
Compat `st,stm32h7-spi` is used on all STM32 that have the same SPI IP than H7 and using this compat avoids dealing with various architectures instead. This is IMO the beast way to deal with this.

Reply:
Okay then, still it's repeated few times, should there be a single #define with this on top?

Reply:
Sure

Reply:
Makes sense, will do.

Reply:
Done.